Sources/_CryptoExtras/MLDSA/MLDSA_boring.swift

```diff
@@ -0,0 +1,319 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crypto
+import Foundation
+
+@_implementationOnly import CCryptoBoringSSL
+@_implementationOnly import CCryptoBoringSSLShims
+
+/// Types associated with the ML-DSA-65 algorithm
+@_documentation(visibility: public)
+public enum MLDSA {}
+
+extension MLDSA {
+    /// A ML-DSA-65 private key.
+    public struct PrivateKey: Sendable {
+        fileprivate let pointer: UnsafeMutablePointer<MLDSA65_private_key>
+
+        /// Initialize a ML-DSA-65 private key from a random seed.
+        /// 
+        /// - Throws: ``CryptoMLDSAError/keyGenerationFailure`` if the key could not be generated.
+        public init() throws {
+            self.pointer = UnsafeMutablePointer<MLDSA65_private_key>.allocate(capacity: 1)
+
+            let publicKeyPtr = UnsafeMutablePointer<UInt8>.allocate(capacity: MLDSA.PublicKey.bytesCount)
+            let seedPtr = UnsafeMutablePointer<UInt8>.allocate(capacity: MLDSA.seedSizeInBytes)
+            defer {
+                publicKeyPtr.deallocate()
+                seedPtr.deallocate()
+            }
+
+            guard CCryptoBoringSSL_MLDSA65_generate_key(
+                publicKeyPtr,
+                seedPtr,
+                self.pointer
+            ) == 1 else {
+                throw CryptoMLDSAError.keyGenerationFailure
+            }
+        }
+
+        /// Initialize a ML-DSA-65 private key from a seed.
+        /// 
+        /// The seed must be at least 32 bytes long.
+        /// Any additional bytes in the seed are ignored.
+        /// 
+        /// - Parameter seed: The seed to use to generate the private key.
+        /// 
+        /// - Throws: `CryptoKitError.incorrectKeySize` if the seed is not at least 32 bytes long or ``CryptoMLDSAError/keyGenerationFailure`` if the key could not be generated.
+        public init(from seed: some DataProtocol) throws {
+            guard seed.count >= MLDSA.seedSizeInBytes else {
+                throw CryptoKitError.incorrectKeySize
+            }
+
+            let seedPtr = UnsafeMutablePointer<UInt8>.allocate(capacity: MLDSA.seedSizeInBytes)
+            seedPtr.initialize(from: seed.regions.flatMap { $0 }, count: MLDSA.seedSizeInBytes)
```

Comment:
There's a [`copyBytes(to:)`](https://developer.apple.com/documentation/foundation/dataprotocol/3139158-copybytes) function we can use for this.